lib/TH/THGeneral.h.in

```diff
@@ -46,6 +46,14 @@
 #define TH_INDEX_BASE 1
 #endif
 
+#ifndef TH_GENERIC_USE_HALF
+# define TH_GENERIC_USE_HALF 0
+#endif
+
+#ifndef TH_NATIVE_HALF
+# define TH_NATIVE_HALF 0
```

Comment:
... and I am also looking for a way to adopt __fp16 (via struct) to be usable as it is, too.